/*
 * This file was automatically generated by EvoSuite
 * Mon Mar 05 22:30:11 GMT 2018
 */

package java_programs;

import org.junit.Test;
import static org.junit.Assert.*;
import java_programs.KNAPSACK;

public class KNAPSACK_ESTest {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      KNAPSACK kNAPSACK0 = new KNAPSACK();
      int int0 = 912;
      int[][] intArrayArray0 = null;
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack(912, (int[][]) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      KNAPSACK kNAPSACK0 = new KNAPSACK();
      int[][] intArray0 = new int[1][3];
      int[] intArray1 = new int[6];
      intArray1[0] = 853;
      intArray1[1] = 853;
      intArray1[2] = 853;
      intArray1[3] = 853;
      intArray1[4] = 853;
      int int0 = 1;
      intArray1[5] = 1;
      intArray0[0] = intArray1;
      KNAPSACK.knapsack(853, intArray0);
      KNAPSACK.knapsack(853, intArray0);
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack((-1), intArray0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // -1
         //
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      int[][] intArray0 = new int[2][5];
      int[] intArray1 = new int[2];
      intArray1[0] = 0;
      intArray1[1] = 0;
      intArray0[0] = intArray1;
      int[] intArray2 = new int[0];
      intArray0[1] = intArray2;
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack(0, intArray0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 0
         //
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      KNAPSACK kNAPSACK0 = new KNAPSACK();
      int[][] intArray0 = new int[2][7];
      int[] intArray1 = new int[2];
      intArray1[0] = 2855;
      intArray1[1] = 2855;
      intArray0[0] = intArray1;
      int[] intArray2 = new int[1];
      intArray2[0] = 2855;
      intArray0[1] = intArray2;
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack(2855, intArray0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 1
         //
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      KNAPSACK kNAPSACK0 = new KNAPSACK();
      int[][] intArray0 = new int[3][0];
      int[] intArray1 = new int[4];
      intArray1[0] = 0;
      intArray1[1] = 1712;
      intArray1[2] = 1712;
      intArray1[3] = 1712;
      intArray0[0] = intArray1;
      int[] intArray2 = new int[3];
      intArray2[0] = 1712;
      intArray2[1] = 0;
      intArray2[2] = 0;
      intArray0[1] = intArray2;
      int[] intArray3 = new int[1];
      intArray3[0] = 1712;
      intArray0[2] = intArray3;
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack(1712, intArray0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 1
         //
      }
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      int[][] intArray0 = new int[3][9];
      int[] intArray1 = new int[8];
      intArray1[0] = 2134;
      intArray1[1] = 2134;
      intArray1[2] = 2134;
      intArray1[3] = 0;
      intArray1[4] = 0;
      intArray1[5] = 2134;
      intArray1[6] = 2134;
      intArray1[7] = 2134;
      intArray0[0] = intArray1;
      int[] intArray2 = new int[9];
      intArray2[0] = 0;
      intArray2[1] = 2134;
      intArray2[2] = 2134;
      intArray2[3] = 0;
      intArray2[4] = 2134;
      intArray2[5] = 0;
      intArray2[6] = 0;
      intArray2[7] = 0;
      intArray2[8] = 2134;
      intArray0[1] = intArray2;
      int[] intArray3 = new int[2];
      intArray3[0] = 0;
      intArray3[1] = 0;
      intArray0[2] = intArray3;
      KNAPSACK.knapsack(2134, intArray0);
      KNAPSACK kNAPSACK0 = new KNAPSACK();
      // Undeclared exception!
      KNAPSACK.knapsack(2134, intArray0);
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      int[][] intArray0 = new int[0][2];
      KNAPSACK.knapsack(151, intArray0);
  }

  @Test(timeout = 4000)
  public void test7()  throws Throwable  {
      KNAPSACK kNAPSACK0 = new KNAPSACK();
      int[][] intArray0 = new int[0][5];
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack((-2017), intArray0);
        fail("Expecting exception: NegativeArraySizeException");
      
      } catch(NegativeArraySizeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }
}
